v2/boost-sdk/components/ManagedBudget/managedBudget.mdx

```diff
@@ -0,0 +1,197 @@
+---
+title: "ManagedBudget"
+---
+import Config from '/snippets/config.mdx';
+import ReturnValue from '/snippets/ReturnValue.mdx'
+
+The ManagedBudget is a flexible budget implementation that allows for the distribution of tokens (ERC-20 and native) with role-based access control. A Managed Budget is required to deploy boosts.
+
+<Note>
+  For a detailed technical overview of the ManagedBudget class, please refer to the [ManagedBudget TypeDoc](https://boost-protocol.vercel.app/classes/ManagedBudget.html).
+</Note>
+
+## Overview
+
+ManagedBudget is designed to hold and distribute assets while providing fine-grained control over who can perform various operations. It supports ETH, ERC-20 based assets.
+
+## Key Features
+
+- Role-based access control for management and administration
+- Support for multiple asset types (ETH, ERC-20)
+- Flexible allocation and distribution of assets
+- Clawback functionality for asset recovery
+
+## Usage
+
+#### Create a new ManagedBudget
+
+**To initialize a new ManagedBudget contract:**
+
+<CodeGroup>
+  ```ts index.ts
+  import { BoostCore } from '@boostxyz/sdk/BoostCore'
+  import { BoostRegistry } from '@boostxyz/sdk/BoostRegistry'
+  import { ManagedBudgetRoles } from '@boostxyz/sdk/Budgets/ManagedBudget'
+  import { config } from "./config";
+
+  const core = new BoostCore({ config });
+  const registry = new BoostRegistry({ config });
+
+  // initialize a new budget contract
+  const budget = await registry.initialize(
+    "MyBoostBudget",
+    core.ManagedBudget({
+      owner: account.address,
+      authorized: [account.address, core.assertValidAddress()],
+      roles: [ManagedBudgetRoles.ADMIN, ManagedBudgetRoles.MANAGER],
+    })
+  );
+
+  const budgetAddress = budget.assertValidAddress();
+  ```
+  <Config />
+</CodeGroup>
+
+#### Get an existing ManagedBudget
+
+**To get an existing ManagedBudget contract:**
+
+<CodeGroup>
+```ts index.ts
+const budget = core.ManagedBudget("0xc55F719709bDad022B320E76f9DfF7e6F5680767")
+
+// or if you want a budget from a specific chain
+const budgetOnBase = core.ManagedBudget(
+  "0xc55F719709bDad022B320E76f9DfF7e6F5680767", 
+  { chainId: 8453 }
+)
+```
+</CodeGroup>
+
+<Tip>
+  [Learn more about targeting specific chains](/v2/boost-sdk/quick-start#targeting-specific-chains)
+</Tip>
+
+### Write Methods
```

Comment:
for consistency, let's merge write and read

### API

#### allocate (same case as in source code for public methods)

Reply:
Was an old draft. Will update it to be more inline with the other pages.